code/__HELPERS/icons.dm

```diff
@@ -692,7 +692,18 @@ The _flatIcons list is a cache for generated icon files.
 	var/image/copy
 	// Add the atom's icon itself, without pixel_x/y offsets.
 	if(!noIcon)
-		copy = image(icon=curicon, icon_state=curstate, layer=A.layer, dir=curdir)
+		if(curdir == SOUTH)
+			copy = image(curicon, curstate)
+		else
+			var/icon/dirIcon
+			if(curdir % 2 == 0)

```

Comment:
Start a test project and verify for yourself. Cardinals are 1 2 4 8 and the diagonals are sums of these values.